client/src/components/layout/ShareTimeTable.jsx

```diff
@@ -1,6 +1,232 @@
+import "../../styles/HelForm.css";
+import { editTT } from "../../actions/UpdateTimeTable";
+
+const branches = [
+  { value: "BIO", label: "Biological Sciences" },
+  { value: "CHE", label: "Chemical Engineering" },
+  { value: "CHEM", label: "Chemistry" },
+  { value: "CE", label: "Civil Engineering" },
+  { value: "CS", label: "Computer Science" },
+  { value: "ECO", label: "Economics & Finance" },
+  { value: "ECE", label: "Electrical & Communication Engineering" },
+  { value: "EEE", label: "Electrical & Electronics Engineering" },
+  { value: "INSTR", label: "Electronics & Instrumentation Engineering" },
+  { value: "MANU", label: "Manufacturing Engineering" },
+  { value: "MATH", label: "Mathematics" },
+  { value: "ME", label: "Mechanical Engineering" },
+  { value: "PHA", label: "Pharmacy" },
+  { value: "PHY", label: "Physics" }
+];
+
+const ShareTimeTable = props => {
+  const [formData, setFormData] = useState({
+    branch: [],
+    year: "",
+    TTs: []
+  });
+
+  const { branch, year, TTs } = formData;
+
+  const submitToMongo = (branch, year) => {
+    let TTData = [];
+    try {
+      let br = [];
+      branch.forEach(item => {
+        br.push(item["value"]);
+      });
+      axios
+        .get("/api/share/sharett", {
+          params: {
+            branch: br,
+            year: year
+          }
+        })
+        .then(response => {
+          console.log("Here");
+          if (response.status !== 200) {
+            throw new Error("Could not submit query.");
+          } else {
+            TTData = JSON.parse(JSON.stringify(response["data"]));
+            setFormData({ ...formData, TTs: TTData });
+          }
+        });
+    } catch (err) {
+      window.alert(err.message);
+    }
+  };
+
+  const handleBranchChange = newBranch => {
+    console.log(newBranch);
+    setFormData({
+      ...formData,
+      branch: newBranch
+    });
+  };
+  const handleYearChange = e => {
+    if (e.target.checked) {
+      setFormData({
+        ...formData,
+        year: e.target.value
+      });
+    }
+  };
+
+  const onSubmit = async e => {
+    e.preventDefault();
+    console.log(year);
+    if (!branch || branch.length === 0) {
+      window.alert("Please enter your branch");
+    } else if (year === "" || year === NaN) {
+      window.alert("Please enter your year");
+    } else {
+      submitToMongo(branch, year);
+    }
+  };
+
+  const Menu = props => {
+    const optionSelectedLength = props.getValue().length || 0;
+    return (
+      <components.Menu {...props}>
+        {optionSelectedLength < 2 ? (
+          props.children
+        ) : (
+          <div className='wide-menu-row'>
+            You cannot select more than 2 branches
+          </div>
+        )}
+      </components.Menu>
+    );
+  };
+  const isValidNewOption = (inputValue, selectValue) =>
+    inputValue.length > 0 && selectValue.length < 5;
+
+  const [userInfo, loading] = useGetData("/api/share/shareTT/");
```

Comment:
```suggestion
  const [userInfo, loading] = useGetData("/api/share/shareTT");
```

Was the last slash required?

Reply:
removed